pkg/controller/command/presentproof/command.go

```diff
@@ -0,0 +1,417 @@
+	SendRequestPresentationErrorCode
+	// AcceptRequestPresentationErrorCode is for failures in accept request presentation command
+	AcceptRequestPresentationErrorCode
+	// NegotiateRequestPresentationErrorCode is for failures in negotiate request presentation command
+	NegotiateRequestPresentationErrorCode
+	// DeclineRequestPresentationErrorCode is for failures in decline request presentation command
+	DeclineRequestPresentationErrorCode
+	// SendProposePresentationErrorCode is for failures in send propose presentation command
+	SendProposePresentationErrorCode
+	// AcceptProposePresentationErrorCode is for failures in accept propose presentation command
+	AcceptProposePresentationErrorCode
+	// DeclineProposePresentationErrorCode is for failures in decline propose presentation command
+	DeclineProposePresentationErrorCode
+	// AcceptPresentationErrorCode is for failures in accept presentation command
+	AcceptPresentationErrorCode
+	// DeclinePresentationErrorCode is for failures in decline presentation command
+	DeclinePresentationErrorCode
+)
+
+const (
+	// command name
+	commandName = "presentproof"
+
+	actions                      = "Actions"
+	sendRequestPresentation      = "SendRequestPresentation"
+	acceptRequestPresentation    = "AcceptRequestPresentation"
+	negotiateRequestPresentation = "NegotiateRequestPresentation"
+	declineRequestPresentation   = "DeclineRequestPresentation"
+	sendProposePresentation      = "SendProposePresentation"
+	acceptProposePresentation    = "AcceptProposePresentation"
+	declineProposePresentation   = "DeclineProposePresentation"
+	acceptPresentation           = "AcceptPresentation"
+	declinePresentation          = "DeclinePresentation"
+)
+
+const (
+	// error messages
+	errEmptyPIID                = "empty PIID"
+	errEmptyMyDID               = "empty MyDID"
+	errEmptyTheirDID            = "empty TheirDID"
+	errEmptyPresentation        = "empty Presentation"
+	errEmptyProposePresentation = "empty ProposePresentation"
+	errEmptyRequestPresentation = "empty RequestPresentation"
+
+	// log constants
+	successString = "success"
+)
+
+var logger = log.New("aries-framework/controller/presentproof")
+
+// Command is controller command for present proof
+type Command struct {
+	client *presentproof.Client
+}
+
+// New returns new present proof controller command instance
+func New(ctx presentproof.Provider) (*Command, error) {
+	client, err := presentproof.New(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create a client: %w", err)
+	}
+
+	// creates action channel
+	actions := make(chan service.DIDCommAction)
+	// registers action channel to listen for events
+	if err := client.RegisterActionEvent(actions); err != nil {
+		return nil, fmt.Errorf("register action event: %w", err)
+	}
+
+	// this code listens for the action events and does nothing
+	// this trick is used to avoid error such as `no clients are registered to handle the message`
```

Comment:
@rolsonquadras @llorllale

Reply:
Is this an architecture design question? Should protocol services require registered listeners for `DIDCommAction` events when handling inbound messages?

Reply:
now, I am having trouble with that design in my next PR. I suggest removing that restriction `no clients are registered to handle the message`.  We do not need a client to be registered bc we can continue at any time by using rest. @troyronda @llorllale Do you agree?

Reply:
This trick doesn't work anyway.  I forgot that we can register only one channel.

Reply:
Or is it the idea? I mean that only one consumer can be registered.

Reply:
Note - DIDExchange sends action event only when clients are registered. https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/service.go#L360